libobs/util/hash/README.libobs

```diff
@@ -0,0 +1,31 @@
+# `uthash`
+
+Vendored from https://github.com/troydhanson/uthash for use in OBS.
+
+Modifications inlucde:
+ - Removed unused files
+ - Formatted with OBS clang-format settings
```

Comment:
We should stop including deps in-tree and rely more on obs-deps, Flatpak modules and Ubuntu packages when possible.

Ubuntu 20.04 does not have a recent enough `uthash-dev` package. If it becomes a blocker for improvements I guess 20.04 support should be dropped.

Reply:
Based on off-chain discussions we believe that right now vendoring is the simplest way forward, and means not having to worry about package availability in Linux distributions and not having to write finders etc. for CMake.

Once the CMake rework for Windows and Linux are finalised and merged and we drop support for 20.04 we can revisit that and move the headers over to obs-deps.